pyomo/contrib/doe/DoE_package.py

```diff
@@ -1596,122 +1488,296 @@ def __get_solver_info(self):
+        ax.set_xlabel(xlabel_text)
+        plt.title(title_text + ' - D optimality')
+        plt.show()
+
+        # Draw E-optimality
+        fig = plt.figure()
+        plt.rc('axes', titlesize=font_axes)
+        plt.rc('axes', labelsize=font_axes)
+        plt.rc('xtick', labelsize=font_tick)
+        plt.rc('ytick', labelsize=font_tick)
+        ax = fig.add_subplot(111)
+        params = {'mathtext.default': 'regular'}
+        # plt.rcParams.update(params)
+        ax.plot(x_range, y_range_E)
+        ax.scatter(x_range, y_range_E)
+        ax.set_ylabel('$log_{10}$ Minimal eigenvalue')
+        ax.set_xlabel(xlabel_text)
+        plt.title(title_text + ' - E optimality')
+        plt.show()
+
+        # Draw Modified E-optimality
+        fig = plt.figure()
+        plt.rc('axes', titlesize=font_axes)
+        plt.rc('axes', labelsize=font_axes)
+        plt.rc('xtick', labelsize=font_tick)
+        plt.rc('ytick', labelsize=font_tick)
+        ax = fig.add_subplot(111)
+        params = {'mathtext.default': 'regular'}
+        # plt.rcParams.update(params)
+        ax.plot(x_range, y_range_ME)
+        ax.scatter(x_range, y_range_ME)
+        ax.set_ylabel('$log_{10}$ Condition number')
+        ax.set_xlabel(xlabel_text)
+        plt.title(title_text + ' - Modified E optimality')
+        plt.show()
 
-    def heatmap(self, title_text, xlabel_text, ylabel_text, font_axes=16, font_tick=14, log_scale=True):
+    def __heatmap(self, title_text, xlabel_text, ylabel_text, font_axes=16, font_tick=14, log_scale=True):
         '''
-        draw heatmaps of the three criteria
+        Draw 2D heatmaps for all design criteria
 
         Parameters:
-        -----------
-        title_text: name of the heatmap, a string
-        xlabel_text: x label title, a string, should be the second design varialbe in the dv_ranges
-        ylabel_text: y label title, a string, should be the first design variable in the dv_ranges
-        font_axes: axis font size
-        font_tick: axis tick font size
+        ----------
+        title_text: name of the figure, a string
+        xlabel_text: x label title, a string.
+            In a 2D heatmap, it should be the second design varialbe in the design_ranges
+        ylabel_text: y label title, a string.
+            In a 2D heatmap, it should be the first design variable in the dv_ranges
+        font_axes: axes label font size
+        font_tick: tick label font size
         log_scale: if True, the result matrix will be scaled by log10
 
-        Note:
-        ----
-        Return 4 heatmaps for A, D, E, Modified E-criteria
+        Returns:
+        --------
+        4 Figures of 2D heatmap for each criteria
         '''
-        # Get meshgrids of design variables and results for plotting
 
-        self.__extract_criteria()
+        # achieve the design variable ranges this figure needs
+        x_range = list(set(self.figure_result_data[self.sensitivity_dimension[0]].values.tolist()))
```

Comment:
What if someone runs an N-dimensional sensitivity analysis and wants to generate a sequence of heatmaps? We can talk more during our meeting.

Reply:
Add a warning to user that I renamed these repeated variables